code/game/objects/items.dm

```diff
@@ -53,6 +52,10 @@
 	var/vending_cat = null// subcategory for vending machines.
 	var/list/dynamic_overlay[0] //For items which need to slightly alter their on-mob appearance while being worn.
 
+	var/is_shrapnel = 0
+	var/shrapnel_amount = 0
+	var/shrapnel_type = ""
```

Comment:
Guns do it that way because fuck guns.
Also, no, you wouldn't have to change that other line, but I have no idea how that works as it is now. I guess `new()` has an unlisted built-in `text2path()`.